src/simulator.ts

```diff
@@ -90,11 +91,48 @@ export class Simulator {
     this.cancelToken = true;
   }
 
+  public async telemetry(eventName: string, result: boolean, properties?: { [key: string]: string; }) {
+    const iotHubConnectionString = result ? await Utility.getConnectionString(
```

Comment:
cache the connection string in Simulator class so we don't need to get it everytime